### PR TITLE
docs: improve cloud docs

### PR DESCRIPTION
**Pull request by michaellzc**
wrap up https://github.com/sourcegraph/sourcegraph/pull/39666

## Test plan

n/a docs changes

### CODE REVIEW

File: doc/cloud/index.md [hunk 1]
```diff
@@ -111,13 +111,20 @@ To learn more about how the Sourcegraph team operates managed SMTP internally, r
...
+- Code hosts or user authentication providers running in a private network are not yet supported. They have to be publically available or they must allow incoming traffic from Sourcegraph-owned static IP addresses. We do not have proper support for other connectivity methods, e.g. site-to-site VPN, VPC peering, tunneling.
+- Instances currently run only on Google Cloud Platform in the `us-central1` region. Other regions and cloud providers (such as AWS or Azure) are not yet supported.
 
 ## Security
 
 Your managed instance will be accessible over HTTPS/TLS, provide storage volumes that are encrypted at rest, and have access restricted to only your team through your enterprise VPN and/or internal [SSO (single sign-on provider)](../admin/auth/index.md) of choice.
 
-If you would like your managed instance to be public, we will provide security capabilities from Cloudflare such as WAF and rate-limiting. We will also provide a firewall to protect your instance from malicious traffic.
+For all managed instances, we will provide security capabilities from Cloudflare such as WAF and rate-limiting to protect your instance from malicious traffic.
```
Comment (michaellzc) [inline doc/cloud/index.md:127]: both public and private instances are now protected by CF

File: doc/cloud/index.md [hunk 2]
```diff
@@ -111,17 +111,24 @@ To learn more about how the Sourcegraph team operates managed SMTP internally, r
...
 
-Only essential Sourcegraph personnel will have access to the instance, server, code, and any other sensitive materials, such as tokens or keys. The employees or contractors with access are bound by the same terms as Sourcegraph itself. Learn more in our [network security policies for Sourcegraph Cloud](https://about.sourcegraph.com/security) or [contact us](https://about.sourcegraph.com/contact/sales) with any questions or concerns.
+Only essential Sourcegraph personnel will have access to the instance, server, code, and any other sensitive materials, such as tokens or keys. The employees or contractors with access are bound by the same terms as Sourcegraph itself. Learn more in our [security policies for Sourcegraph Cloud](https://about.sourcegraph.com/security) or [contact us](https://about.sourcegraph.com/contact/sales) with any questions or concerns. You may also request a copy of our SOC 2 Report, please visit our [Security Trust Portal](https://security.sourcegraph.com).
```
Comment (michaellzc) [inline doc/cloud/index.md:131]: I think it's nice to mention we have a SOC2 report on the customer facing site

File: doc/cloud/index.md [hunk 3]
```diff
@@ -46,6 +46,13 @@ As part of this service you will receive a number of benefits from our team, inc
   - [Adding repositories from all of your code hosts to Sourcegraph](../admin/external_service/index.md)
   - [Integrating your single sign-on provider with Sourcegraph](../admin/auth/index.md)
   - [Configuring Sourcegraph](../admin/config/index.md)
+  
+### Access to all Sourcegraph features
+
+All Sourcegraph features are avilable on day-one, such as:
+
+- [Server-side Batch Changes](../batch_changes/explanations/server_side.md)
+- [Precise code navigation powered by auto-indexing](../code_navigation/explanations/auto_indexing.md)
```
Comment (michaellzc) [inline doc/cloud/index.md:55]: there was confusion around whether executors are configure on cloud by default or not

https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1670520292251349

File: doc/cloud/index.md (hunk 2 above)
Comment (sqs) [reply doc/cloud/index.md:139]:
```suggestion
Only essential Sourcegraph personnel will have access to the instance, server, code, and any other sensitive materials, such as tokens or keys. The employees or contractors with access are bound by the same terms as Sourcegraph itself. Learn more in our [security policies for Sourcegraph Cloud](https://about.sourcegraph.com/security) or [contact us](https://about.sourcegraph.com/contact/sales) with any questions or concerns. You may also request a copy of our SOC 2 Report on our [security portal](https://security.sourcegraph.com).
```

**Review by sqs — APPROVED** [review]

File: doc/cloud/index.md [hunk 4]
```diff
@@ -46,6 +46,13 @@ As part of this service you will receive a number of benefits from our team, inc
   - [Adding repositories from all of your code hosts to Sourcegraph](../admin/external_service/index.md)
   - [Integrating your single sign-on provider with Sourcegraph](../admin/auth/index.md)
   - [Configuring Sourcegraph](../admin/config/index.md)
+  
+### Access to all Sourcegraph features
+
+All Sourcegraph features are avilable on day-one, such as:
```
Comment (bobheadxi) [inline doc/cloud/index.md:52]:
```suggestion
All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-box, such as:
```

**Review by bobheadxi — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

**Pull request merged**